src/policy/vmspace.rs

```diff
@@ -0,0 +1,249 @@
+use crate::plan::{CreateGeneralPlanArgs, CreateSpecificPlanArgs};
+use crate::plan::{ObjectQueue, VectorObjectQueue};
+use crate::policy::immortalspace::ImmortalSpace;
+use crate::policy::sft::GCWorkerMutRef;
+use crate::policy::sft::SFT;
+use crate::policy::space::{CommonSpace, Space};
+use crate::util::address::Address;
+use crate::util::heap::HeapMeta;
+use crate::util::heap::PageResource;
+use crate::util::heap::VMRequest;
+use crate::util::metadata::side_metadata::SideMetadataContext;
+use crate::util::metadata::side_metadata::SideMetadataSanity;
+use crate::util::ObjectReference;
+use crate::vm::VMBinding;
+
+pub struct VMSpace<VM: VMBinding> {
+    inner: Option<ImmortalSpace<VM>>,
+    // Save it
+    args: CreateSpecificPlanArgs<VM>,
+}
+
+impl<VM: VMBinding> SFT for VMSpace<VM> {
+    fn name(&self) -> &str {
+        self.space().name()
+    }
+    fn is_live(&self, object: ObjectReference) -> bool {
+        self.space().is_live(object)
+    }
+    fn is_reachable(&self, object: ObjectReference) -> bool {
+        self.space().is_reachable(object)
+    }
+    #[cfg(feature = "object_pinning")]
+    fn pin_object(&self, object: ObjectReference) -> bool {
+        self.space().pin_object(object)
+    }
+    #[cfg(feature = "object_pinning")]
+    fn unpin_object(&self, object: ObjectReference) -> bool {
+        self.space().unpin_object(object)
+    }
+    #[cfg(feature = "object_pinning")]
+    fn is_object_pinned(&self, object: ObjectReference) -> bool {
+        self.space().is_object_pinned(object)
+    }
```

Comment:
If this pattern (`self.space().xxxx()`) repeats, we may introduce the `delegate` crate: https://crates.io/crates/delegate

Reply:
I made the changes as suggested, using the `delegate` crate. It works well.